design-toolkit/components/src/components/hero/index.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { Icon } from '@/index';
+import type { PropsWithChildren, ReactElement } from 'react';
+import { Children, createContext, isValidElement, useContext } from 'react';
+import type { ContextValue } from 'react-aria-components';
+import { type HeroStyleVariants, HeroStyles } from './styles';
+
+interface ChildProps extends PropsWithChildren {
+  className?: string;
+}
+
+interface HeroProps extends HeroStyleVariants, PropsWithChildren {
+  className?: string;
+}
+
+export const HeroContext =
+  createContext<ContextValue<HeroProps, HTMLDivElement>>(null);
+
+export function Hero({ children, className, ...props }: HeroProps) {
+  const slots = {
+    icon: null as ReactElement | null,
+    primary: null as ReactElement | null,
+    secondary: [] as ReactElement[],
+    invalid: [] as ReactElement[],
+  };
+  const styles = HeroStyles({ ...props });
+
+  Children.forEach(children, (child) => {
+    if (isValidElement(child)) {
+      const childType = child.type as { displayName?: string };
+      const displayName = childType?.displayName;
+
+      if (displayName === 'Hero.Icon') {
+        slots.icon = child;
+      } else if (displayName === 'Hero.Primary') {
+        slots.primary = child;
+      } else if (displayName === 'Hero.Secondary') {
+        slots.secondary.push(child);
+      } else {
+        slots.invalid.push(child);
+      }
+    }
+  });
+
+  return (
+    <HeroContext.Provider value={props}>
+      <header className={styles.container({ className })}>
```

Comment:
We can simplify the structure of this since we have two very basic layouts, grid and stack. If you use the layout data attribute, then you can use that selector in the styles for children: https://github.com/gohypergiant/standard-toolkit/blob/main/design-toolkit/components/src/index.css#L42

Reply:
I think I've got this now.